commercetools-models/src/main/java/io/sphere/sdk/products/commands/ProductImageUploadCommand.java

```
@@ -0,0 +1,104 @@
+        return new ProductImageUploadCommand(body, productId, null, sku, null, null);
+    }
+
+    public static ProductImageUploadCommand ofVariantId(final File body, final ByIdVariantIdentifier variantIdentifier) {
+        return new ProductImageUploadCommand(body, variantIdentifier.getProductId(), variantIdentifier.getVariantId(), null, null, null);
+    }
+
+    public ProductImageUploadCommand withFilename(final String newFilename) {
+        return new ProductImageUploadCommand(body, productId, variant, sku, newFilename, staged);
+    }
+
+    public ProductImageUploadCommand withStaged(final Boolean newStaged) {
+        return new ProductImageUploadCommand(body, productId, variant, sku, filename, newStaged);
+    }
+
+    public ProductImageUploadCommand withContentType(final String contentType) {
+        final ProductImageUploadCommand productImageUploadCommand = new ProductImageUploadCommand(body, productId, variant, sku, filename, staged);
+        productImageUploadCommand.givenContentType = contentType;
+        return productImageUploadCommand;
+    }
+
+    @Override
+    protected JavaType jacksonJavaType() {
+        return SphereJsonUtils.convertToJavaType(Product.typeReference());
+    }
+
+    @Override
+    public HttpRequestIntent httpRequestIntent() {
+        final String contentType;
+        if (givenContentType != null) {
+            contentType = givenContentType;
+        } else if (body.getName().toLowerCase().endsWith(".jpeg") || body.getName().toLowerCase().endsWith(".jpg")) {
+            contentType = "image/jpeg";
+        } else if (body.getName().toLowerCase().endsWith(".gif")) {
+            contentType = "image/gif";
+        } else if (body.getName().toLowerCase().endsWith(".png")) {
+            contentType = "image/png";
+        } else {
+            contentType = "unknown";
+            new IllegalStateException("cannot determine content type for " + body.getName());
```

Here we should just throw the exception.